harness/determined/pytorch/_trainer.py

```diff
@@ -147,7 +152,9 @@ def fit(
                 raise ValueError("max_length must be defined in local training mode.")
 
             if not isinstance(max_length.value, int):
-                raise TypeError("max_length must be configured in TrainUnit(int) types.")
+                raise TypeError(
+                    "max_length must either be a det.pytorch.Batch() or det.pytorch.Epoch() type"
+                )
```

Comment:
random question, though not necessarily part of this change:

Wouldn't it be better to do: `if not isinstance(max_length, (Batch, Epoch))`?  The error message is like you're testing `max_length` but the check is actually applying to `max_length.value`.

Also below in the backwards compatibility checks.

Reply:
sort of, there's two separate checks here that i'm confusing. 1 that max length is of Batch/Epoch class, 2 that what was passed into Batch/Epoch is an int (and not a collection, which we support for validation/checkpoint periods). 

so i combined these two checks into a combined error message.